src/convertToObject.js

```diff
@@ -5,8 +5,30 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const delEmptyLine = sourceString.replace(/\n/g, '');
+
+  const commandsArr = delEmptyLine.split(';');
+
+  const commandsArrFinal = commandsArr.filter((str) => str.trim());
+
+  const resultObject = {};
+
+  for (let i = 0; i < commandsArrFinal.length; i++) {
+    const commandArr = commandsArrFinal[i].split(':');
+
+    const key = commandArr[0].trim();
+    const value = commandArr[1].trim();
+
+    resultObject[key] = value.replace(/,/g, ',\n');
+  }
+
+  if (sourceString === '') {
+    return {};
+  }
```

Comment:
[CODE KNOWLEDGE] - This check for an empty string is redundant, as the methods used above (split, filter, map/reduce) will handle an empty string appropriately and return an empty object.